src/text-to-code/main.py

```diff
@@ -0,0 +1,79 @@
+"""
+Core Text to Code code. This is the shared logic used for both the API/demo site and the Lambda.
+"""
+
+
+class SchematronError:
+    location: str
+
+
+class Eicr:
+    pass
+
+
+class Observation:
+    pass
+
+
+class Vector:
+    pass
+
+
+def resolve_nonstandard_text(schematron_output: str, eicr_id: str) -> str:
+    """Given a schematron output and an eICR return a list of XPaths and new codes.
+
+    Used by the Lambda.
+    """
+
+    errors = _get_relevant_errors(schematron_output)
+    eicr = _retrieve_ecir(eicr_id)
+
+    output = dict()
+
+    for error in errors:
+        embedding_candidates = _get_possibly_relevant_string(error, eicr)
+
+        best_string = _get_best_candidate(embedding_candidates)
+
+        opensearch_result = get_best_match(best_string)
+
+        output[error.location] = opensearch_result
+
+    return opensearch_result
+
+
+def get_best_match(string: str) -> str:
+    """Given a string return the best match.
+
+    Used by the API and demo site.
+    """
+
+    vectorized_input = _vectorize(string)
+
+    return _vector_search(vectorized_input)
+
+    pass
+
+
+def _retrieve_ecir(ecir_id) -> Eicr:
+    pass
```

Comment:
Again, fine to stub out for now, but I would also break this out into a subservice like src/text-to-code/eicr/eicr_service.py and put all the eICR-related code there

Reply:
I think that makes sense to me as well.